core/modules/layout/includes/layout.class.inc

```diff
@@ -691,11 +691,26 @@ class Layout {
             $new_positions[$new_region_name][] = $uuid;
           }
           else {
-            unset($this->content[$uuid]);
+            $orphaned_blocks[] = $uuid;
           }
         }
       }
     }
+    if (!empty($orphaned_blocks)) {
+      // Assign orphan blocks to the default or last region,
+      // if the former doesn't exist.
+      end($new_positions);
+      $default_region = isset($new_layout_info['default region']) ? $new_layout_info['default region'] : key($new_positions);
+      $orphaned_blocks_titles = array();
+      foreach ($orphaned_blocks as $uuid) {
+        $new_positions[$default_region][] = $uuid;
+        $orphaned_blocks_titles[] = $this->content[$uuid]->getAdminTitle();
+      }
+      $message = t('The following blocks have been reassigned to the %new_region region. The regions they were placed in previously do not exist in the new layout template. You can move them to more appropriate regions:', array(
+        '%new_region' => $new_layout_info['regions'][$default_region]));
+      $message .= theme('item_list', array('items' => $orphaned_blocks_titles));
+      backdrop_set_message($message, 'warning');
```

Comment:
Setting a UI message within an API function like this is mixing layers that are meant to be independent. If this were called as an API (such as in an update hook), you might end up with unwanted messages. Rather than setting the message here, could we make a new parameter for `&$orphaned_blocks = array()` that is passed by reference? Then when the form submit handler calls `$layout->setLayoutTemplate()` it would get the list of orphaned blocks and set the message in the submit handler.

Reply:
@quicksketch thanks for the review. I've moved the message to the submit handler. There were two parameters I needed from the class. Instead of passing by reference in the method I added them as parameters to the class. I figured it might be better that way.